### ISSUE TITLE
Understanding training parameters

### ISSUE THREAD

**Issue by sidgan** [davidsandberg/facenet#405]
Parameters for training as given in the file: https://github.com/davidsandberg/facenet/blob/master/src/train_tripletloss.py are: 

Batch size = 90 images
Image size = 160 x 160 x 3
People per batch = 45
Image per person = 40 
Number of batches in one epoch = 1000 

This means that each batch has 90 x 160 x 160 x 3 bytes = 6.591796875 Gigabytes and is dependent on the available GPU memory. This calculation ignores the 'people per batch' and 'images per person' parameters. Can you please explain how these two parameters are tuned and how they are utilized in loading the images in each batch.

**Comment by qixianbiao** [comment]
Hi, 
1:  90 x 160 x 160 x 3 bytes = 6.59MB, 
2: 90*160*160*32 + 90*79*79*64 + 90*79*79*80 + ........ > 5GB
3: Tensorflow does not optimize the usage of memory.  So,  I can not figure out where half of memory (6G) goes.
4:  Parameters are chosen due to memory limitation.

**Comment by davidsandberg** [comment]
Hi,
For training using triplet loss you can have a look at #124.
